src/main/java/Quag.java

```diff
@@ -0,0 +1,150 @@
+        if (parts.length >= 2) {
+            try {
+                int taskIndex = Integer.parseInt(parts[1]);
+                ListUtils.markAsDone(taskIndex);
+            } catch (NumberFormatException e) {
+                System.out.println("That's not a quaggin number!!");
+            }
+        } else {
+            System.out.println("There's no quaggin number!!");
+        }
+    }
+    private static void unmarkTask(String userInput) {
+        String[] parts = userInput.split(" ");
+        if (parts.length >= 2) {
+            try {
+                int taskIndex = Integer.parseInt(parts[1]);
+                ListUtils.markAsNotDone(taskIndex);
+            } catch (NumberFormatException e) {
+                System.out.println("That's not a quaggin number!!");
+            }
+        } else {
+            System.out.println("There's no quaggin number!!");
+        }
+    }
+
+    private static void addTodo(String userInput) {
+        //description is from parts[1] to end
+        String description = userInput.substring("todo".length()).trim();
+        ListUtils.addTodoToList(description);
+    }
+
+    private static void addDeadline(String userInput) {
+        //description is from parts[1] until / is detected
+        //deadline is after /
+
+        // int to denote length and position of "/by" separator
+        int byLength = 4;
+        int byIndex = userInput.indexOf("/by");
+
+        //find position of /by separator
+        if (byIndex != -1) {
+            //extract the description and deadline
+            String description = userInput.substring("deadline".length(), byIndex).trim();
+            // Skip over "/by "
+            String by = userInput.substring(byIndex + byLength).trim();
+            ListUtils.addDeadlineToList(description, by);
+        } else {
+            System.out.println("That's not a valid quaggin format for a deadline!!");
+        }
+
+    }
+
+    private static void addEvent(String userInput) {
+        //description is from parts[1] until first /
+        //from is in between first and second /
+        //to is after second /
+
+        // int to denote length and position of "/from" and "/to" separator
+        int fromIndex = userInput.indexOf("/from");
+        int fromLength = 6;
+        int toIndex = userInput.indexOf("/to");
+        int toLength = 4;
+
+        if (fromIndex != -1 && toIndex != -1 && toIndex > fromIndex) {
+            // extract description, from and to
+            String description = userInput.substring("event".length(), fromIndex).trim();
+            String from = userInput.substring(fromIndex + fromLength, toIndex).trim();
+            String to = userInput.substring(toIndex + toLength).trim();
+            ListUtils.addEventToList(description, from, to);
+        }  else {
+            System.out.println("That's not a valid quaggin format for an event!!");
+        }
+
+    }
+
+    private static void displayCommandList() {
```

Comment:
Here, comment should be self explanatory. For the description part, one can change the function name todisplayCommandListWithDescription or comment can be shortened to just #commands with description (preferably the function name)